### ISSUE TITLE
Refactor legacy Storybook stories

### ISSUE THREAD

**Issue by benbowler** [google/site-kit-wp#9408]
## Feature Description

**Refactor Legacy Stories**

We have a number of Storybook stories in the legacy format, in the folder `/stories/`. These stories lead to inconsistent structure in the way stories show when built. Updating them will make our stories clear and consistent across all modules and components.

Some examples:

1. The Account Chooser and Settings stories, if updated, could show in the global or components group rather than at the root level:
<img width="219" alt="Screenshot 2024-09-23 at 12 23 34" src="https://github.com/user-attachments/assets/52d3ca19-22dd-4c7d-810b-28ecd92544eb">

2. Older modules stories should be updated so that they show in the Modules folder and their stories can be sorted alphabetically: 
<img width="207" alt="Screenshot 2024-09-23 at 12 23 39" src="https://github.com/user-attachments/assets/d4b66909-287a-4013-8594-d858279fe0e8">
<img width="226" alt="Screenshot 2024-09-23 at 12 23 45" src="https://github.com/user-attachments/assets/a862827e-809f-42d3-8d7d-deb3ce24c0a3">
<img width="170" alt="Screenshot 2024-09-23 at 12 23 49" src="https://github.com/user-attachments/assets/5f9384d2-b098-4e5f-a85b-1bfe7f084e8b">

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Legacy Storybook stories (those using the `storiesOf` API) should be refactored to use the modern Component Story Format (aka CSF)
	* Stories should be logically organized together in a way which is consistent with how new stories are organized today
* Stories with related VRT scenarios should maintain the same target scenario files for now (i.e. no changes to VRT should happen as part of this refactoring)
* This work should be limited to the set of stories refactored in #9739 
	* New issues should be created as needed to address the remaining instances using similar criteria as this one
	* Refactoring is completed when all stories under `/stories/` have been refactored and the directory is no longer used or referenced 

## Implementation Brief

* [ ] Finish and merge #9739
* [ ] Resolve use of react-router-dom for the stories in `SettingsModules` refactored story:
* https://github.com/google/site-kit-wp/blob/09cbb3c13a0aa381d0a10ca8a4f7ccfea0c9ab08/assets/js/components/settings/SettingsModules.stories.js#L79-L83

### Test Coverage

* No changes are to be made to Storybook or VRT coverage.

### QA:Eng

* Review storybook, confirm the components whose stories have been changed in this commit still show in the updated storybook (note some stories have changed structure and folder but each component state should still be visible in storybook).

## Changelog entry

* N/A.

**Comment by benbowler** [comment]
Merge conflicts fixed and added an outstanding item to the IB from the hackathon + added estimate.

I also created a follow up issue #9882 which will allow us to complete the refactoring of the legacy module stories as well as the widget stories which I ran out of time with on the Hackathon.

**Comment by techanvil** [comment]
## QA ✅

The stories have all been verified during CR. I tested them on a version of the branch with `develop` merged into it, in order to ensure the testing was performed as it would be on `develop` once the branch is merged.
